src/email/outbound.js

```diff
@@ -0,0 +1,326 @@
+                            log.debug("Server supports STARTTLS, continuing");
+                            w('STARTTLS');
+                            isUpgradeInProgress = true;
+                            break;
+                        }
+                        else {
+                            upgraded = true;
+                            log.debug("No STARTTLS support or ignored, continuing");
+                        }
+                    }
+                    w(queue[step]);
+                    step++;
+                    break;
+
+                case 251: // forward
+                    if (step === queue.length - 1) {
+                        log.info('OK:', code, msg);
+                        return;
+                    }
+                    w(queue[step]);
+                    step++;
+                    break;
+
+                case 354:
+                    // Send message, inform end by `<CR><LF>.<CR><LF>`
+                    log.info('Sending mail body', body);
+                    w(body);
+                    w('');
+                    w('.');
+                    break;
+
+                case 334: // Send login details [for relay]
+                    // TODO: support login.
+                    w(login[loginStep]);
+                    loginStep++;
+                    break;
+
+                default:
+                    if (code >= 400) {
+                        log.error('SMTP server responds with error code', code);
+                        sock.end();
+                        throw Error(`SMTP server responded with code: ${code} + ${msg}`);
+                    }
+            }
+        }
+
+        let msg = '';
+
+        function onLine(line) {
+            log.debug('RECV ' + domain + '>' + line);
+
+            msg += (line + CRLF);
+
+            if (line[3] === ' ') {
+                // 250-information dash is not complete.
+                // 250 OK. space is complete.
+                let lineNumber = parseInt(line.substr(0, 3));
+                response(lineNumber, msg);
+                msg = '';
+            }
+        }
+    }
+
+    /**
+     * Send Mail directly
+     * `mail` object attribute reference: https://nodemailer.com/extras/mailcomposer/#e-mail-message-fields
+     * @param mail {object}
+     *             from
+     *             to
+     *             cc
+     *             bcc
+     *             replyTo
+     *             returnTo
+     *             subject
+     *             type         default 'text/plain', 'text/html'
+     *             charset      default 'utf-8'
+     *             encoding     default 'base64'
+     *             id           default timestamp+from
+     *             headers      object
+     *             content
+     *             attachments
+     *               [{
+     *                 type
+     *                 filename
+     *                 content
+     *               }].
+     *
```

Comment:
This might be a bit too noisy for a docstring, if the link you've provided above does the same thing?